src/Microsoft.TestPlatform.Extensions.HtmlLogger/HtmlLogger.cs

```diff
@@ -122,6 +122,9 @@ public void Initialize(TestLoggerEvents events, string testResultsDirPath)
             TestRunDetails = new TestRunDetails();
             Results = new ConcurrentDictionary<Guid, ObjectModel.TestResult>();
             ResultCollectionDictionary = new ConcurrentDictionary<string, TestResultCollection>();
+
+            // Ensure test results directory exists.
+            Directory.CreateDirectory(testResultsDirPath);
```

Comment:
That would just normally error out. What do you suggest?

Reply:
If exception message is not user friendly, maybe throw new exception with more useful information, in other case that should be ok.

Reply:
Agreed with @nohwnd, there's not much we can do here if we don't have write access. It's an exceptional situation as are the sanity checks at the start of the method. The only thing that comes to mind is maybe throwing some nicer exception message instead of the default one, as @Sanan07 did in [this](https://github.com/microsoft/vstest/commit/ef615ae48aa04a90dc98071cc82893c849dde97c) PR.

Also, maybe we should be checking if the directory exists first before going ahead and creating it ?

Reply:
We frankly don't even have a repro of this happening, just a report that sometimes the folder does not exist, for some reason. But everything suggests that this folder should be there already, so this is mostly just a sanity check than anything else. I would go with the custom error message route once we get that reported, as it would require add more resource strings for the same thing again.

k?